### PR TITLE
Xenial

### PR DESCRIPTION
**Pull request by mestery**

### CODE REVIEW

File: ci/prepare.sh [hunk 1]
```diff
@@ -14,7 +14,11 @@ sudo apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58
 
 if [ "$INSTALLDOCKER" == "True" ] ; then
     if [ ! -f /etc/apt/sources.list.d/docker.list ] ; then
-        sudo su -c 'echo "deb https://apt.dockerproject.org/repo ubuntu-trusty main" > /etc/apt/sources.list.d/docker.list'
+        if [ "$UBUNTUVER" == "14.04" ]; then
+            sudo su -c 'echo "deb https://apt.dockerproject.org/repo ubuntu-trusty main" > /etc/apt/sources.list.d/docker.list'
+        elif [ "$UBUNTUVER" == "16.04" ]; then
+            sudo su -c 'echo "deb https://apt.dockerproject.org/repo ubuntu-xenial main" > /etc/apt/sources.list.d/docker.list'
+        fi
```
Comment (huikang) [inline ci/prepare.sh]: @mestery to be consistent, can we use `case $UBUNTUVER` as the code below? Thanks.

Comment (mestery) [reply ci/prepare.sh]: Got it, I'll modify this to be a "case" statement instead of "if ; elif ; fi".

File: ci/prepare.sh [hunk 2]
```diff
@@ -133,7 +143,17 @@ sudo /etc/init.d/ssh restart
...
-    sudo service docker start
+    case $UBUNTUVER in
+    14.04)
+        $OVNSUDO service docker start
+        ;;
+    16.04)
+        $OVNSUDO systemctl restart docker
+        ;;
+    *)
+        echo "Warning: Ubuntu version $UBUNTUVER not supported"
```
Comment (huikang) [inline ci/prepare.sh:154]: Is "return 1" missing here?

Comment (mestery) [reply ci/prepare.sh:154]: D'oh, yes it is! New version inbound.

File: ci/prepare.sh [hunk 3]
```diff
@@ -14,7 +14,17 @@ sudo apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58
 
 if [ "$INSTALLDOCKER" == "True" ] ; then
     if [ ! -f /etc/apt/sources.list.d/docker.list ] ; then
-        sudo su -c 'echo "deb https://apt.dockerproject.org/repo ubuntu-trusty main" > /etc/apt/sources.list.d/docker.list'
+        case $UBUNTUVER in
+            14.04)
+                sudo su -c 'echo "deb https://apt.dockerproject.org/repo ubuntu-trusty main" > /etc/apt/sources.list.d/docker.list'
```
Comment (huikang) [inline ci/prepare.sh]: nitpick: `sudo` -> `$OVNSUDO`

File: ci/prepare.sh [hunk 4]
```diff
@@ -14,7 +14,17 @@ sudo apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58
...
+                ;;
+            16.04)
+                sudo su -c 'echo "deb https://apt.dockerproject.org/repo ubuntu-xenial main" > /etc/apt/sources.list.d/docker.list'
```
Comment (huikang) [inline ci/prepare.sh]: ditto

File: ci/prepare.sh (hunk 3 above)
Comment (mestery) [reply ci/prepare.sh]: Done

File: ci/prepare.sh [hunk 5]
```diff
@@ -14,7 +14,17 @@ sudo apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58
...
+                ;;
+            *)
+                echo "Warning: Ubuntu version $UBUNTUVER not supported"
+                return 1
+        esac
         sudo apt-get update -y
         sudo apt-get purge -y lxc-docker
         sudo apt-get install -y apparmor
```
Comment (huikang) [inline ci/prepare.sh]: ditto

File: ci/prepare.sh (hunk 4 above)
Comment (mestery) [reply ci/prepare.sh]: Done

File: ci/prepare.sh [hunk 6]
```diff
@@ -14,7 +14,17 @@ sudo apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58
...
+                $OVNSUDO su -c 'echo "deb https://apt.dockerproject.org/repo ubuntu-trusty main" > /etc/apt/sources.list.d/docker.list'
+                ;;
+            16.04)
+                $OVNSUDO su -c 'echo "deb https://apt.dockerproject.org/repo ubuntu-xenial main" > /etc/apt/sources.list.d/docker.list'
+                ;;
+            *)
+                echo "Warning: Ubuntu version $UBUNTUVER not supported"
+                return 1
+        esac
         sudo apt-get update -y
```
Comment (huikang) [inline ci/prepare.sh]: sudo -> $OVNSUDO

File: ci/prepare.sh (hunk 5 above)
Comment (mestery) [reply ci/prepare.sh]: Done

**Comment by huikang** [comment]
@mestery thanks!

**Pull request merged**